pkg/controller/pxcbackup/controller.go

```diff
@@ -524,76 +660,184 @@ func (r *ReconcilePerconaXtraDBClusterBackup) updateJobStatus(bcp *api.PerconaXt
+	since := time.Since(cr.CreationTimestamp.Time).Seconds()
+
+	if cr.Spec.PassiveDeadlineSeconds == nil {
+		return since, nil
+	}
+
+	if since < float64(*cr.Spec.PassiveDeadlineSeconds) {
+		return since, nil
+	}
+
+	switch cr.Status.State {
+	case api.BackupNew:
+		return since, errors.New("passive deadline seconds exceeded")
+	default:
+		return since, nil
 	}
-	obj.SetOwnerReferences(append(obj.GetOwnerReferences(), ownerRef))
-	return nil
 }
 
-func (r *ReconcilePerconaXtraDBClusterBackup) isOtherBackupRunning(ctx context.Context, cr *api.PerconaXtraDBClusterBackup, cluster *api.PerconaXtraDBCluster) (bool, error) {
-	list := new(batchv1.JobList)
-	if err := r.client.List(ctx, list, &client.ListOptions{
-		Namespace:     cluster.Namespace,
-		LabelSelector: labels.SelectorFromSet(naming.LabelsBackup(cluster)),
-	}); err != nil {
-		return false, errors.Wrap(err, "list jobs")
+func (r *ReconcilePerconaXtraDBClusterBackup) updateStatus(ctx context.Context, cr *api.PerconaXtraDBClusterBackup) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		localCr := new(api.PerconaXtraDBClusterBackup)
+		err := r.client.Get(ctx, client.ObjectKeyFromObject(cr), localCr)
+		if err != nil {
+			return err
+		}
+
+		localCr.Status = cr.Status
+
+		return r.client.Status().Update(ctx, localCr)
+	})
+}
+
+func (r *ReconcilePerconaXtraDBClusterBackup) suspendJobIfNeeded(
+	ctx context.Context,
+	cr *api.PerconaXtraDBClusterBackup,
+	cluster *api.PerconaXtraDBCluster,
+) error {
+	if cluster.Spec.Unsafe.BackupIfUnhealthy {
+		return nil
+	}
+
+	if cluster.Status.Status == api.AppStateReady {
+		return nil
 	}
 
-	for _, job := range list.Items {
-		backupNameLabelKey := naming.LabelPerconaBackupName
-		if cluster.CompareVersionWith("1.16.0") < 0 {
-			backupNameLabelKey = "backup-name"
+	if cluster.Status.PXC.Ready == cluster.Status.PXC.Size {
+		return nil
+	}
+
+	log := logf.FromContext(ctx)
+
+	labelKeyBackupType := naming.GetLabelBackupType(cluster)
+	jobName := naming.BackupJobName(cr.Name, cr.Labels[labelKeyBackupType] == "cron")
+
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		job := new(batchv1.Job)
+
+		err := r.client.Get(ctx, types.NamespacedName{Namespace: cr.Namespace, Name: jobName}, job)
+		if err != nil {
+			if k8sErrors.IsNotFound(err) {
+				return nil
+			}
+			return err
 		}
-		if job.Labels[backupNameLabelKey] == cr.Name {
-			continue
+
+		suspended := false
+		for _, cond := range job.Status.Conditions {
+			if cond.Type == batchv1.JobSuspended && cond.Status == corev1.ConditionTrue {
+				suspended = true
+			}
 		}
-		if job.Status.Active == 0 && (jobSucceded(&job) || jobFailed(&job)) {
-			continue
+
+		if suspended {
+			return nil
 		}
 
-		return true, nil
-	}
+		log.Info("Suspending backup job",
+			"job", jobName,
+			"clusterStatus", cluster.Status.Status,
+			"readyPXC", cluster.Status.PXC.Ready)
+
+		t := true
```

Comment:
I checked a bit and see that we use this variable in several places. 
Maybe it will be a bit nice define 

```
func Bool(b bool) *bool {
    return &b
}
job.Spec.Suspend = Bool(true)/Bool(false)
```

Reply:
we can use `ptr.To` from k8s.io/utils/ptr

Reply:
yep! it also looks nice for me
job.Spec.Suspend = ptr.To(true)